local_server/app/app.py

```diff
@@ -0,0 +1,225 @@
+import datetime
+import logging
+from functools import wraps
+from http import HTTPStatus
+
+from flask import (
+    Flask,
+    current_app,
+    make_response,
+    render_template,
+    Blueprint,
+    request,
+    send_from_directory,
+)
+from flask_restful import Api, Resource
+
+import local_server.common.rest as common_rest
+from local_server.common.errors import DatasetAccessError, RequestException
+from local_server.common.health import health_check
+from local_server.common.utils.utils import Float32JSONEncoder
+
+webbp = Blueprint("webapp", "local_server.common.web", template_folder="templates")
+
+
+@webbp.route("/", methods=["GET"])
+def dataset_index():
+    app_config = current_app.app_config
+
+    dataset_config = app_config.get_dataset_config()
+    scripts = dataset_config.app__scripts
+    inline_scripts = dataset_config.app__inline_scripts
+
+    try:
+        args = {"SCRIPTS": scripts, "INLINE_SCRIPTS": inline_scripts}
+        return render_template("index.html", **args)
+
+    except DatasetAccessError as e:
+        return common_rest.abort_and_log(
+            e.status_code, f"Invalid dataset: {e.message}", loglevel=logging.INFO, include_exc_info=True
+        )
+
+
+@webbp.errorhandler(RequestException)
+def handle_request_exception(error):
+    return common_rest.abort_and_log(error.status_code, error.message, loglevel=logging.INFO, include_exc_info=True)
+
+
+def requires_authentication(func):
+    @wraps(func)
+    def wrapped_function(self, *args, **kwargs):
+        auth = current_app.auth
+        if auth.is_user_authenticated():
+            return func(self, *args, **kwargs)
+        else:
+            return make_response("not authenticated", HTTPStatus.UNAUTHORIZED)
+
+    return wrapped_function
+
+
+def rest_get_data_adaptor(func):
+    @wraps(func)
+    def wrapped_function(self):
+        try:
+            return func(self, current_app.data_adaptor)
+        except DatasetAccessError as e:
+            return common_rest.abort_and_log(
+                e.status_code, f"Invalid dataset: {e.message}", loglevel=logging.INFO, include_exc_info=True
+            )
+
+    return wrapped_function
+
+
+class HealthAPI(Resource):
+    def get(self):
+        config = current_app.app_config
+        return health_check(config)
+
+
+class SchemaAPI(Resource):
+    # TODO @mdunitz separate dataset schema and user schema
```

Comment:
@mckinsel Sorry, I have a rather annoying request.. for each of the TODOs in the comments in this PR (I count 9 I think), would you mind creating a ticket for it and tagging it with the "Dev-ish" label? That way we don't lose the todos in the code forever.

Reply:
Yes 😐